### ISSUE TITLE
Winbar conflicts with Telescope

### ISSUE THREAD

**Issue by ChrisAmelia** [neovim/neovim#18733]
### Neovim version (nvim -v)

v0.8.0-dev+296-g307c5c63e

### Operating system/version

Ubuntu 18

### Terminal name/version

bash

### $TERM environment variable

xterm-kitty

### Installation

appimage

### How to reproduce the issue

**init.vim**

```vim
set runtimepath^=~/.local/share/nvim/site/pack/packer/opt/telescope.nvim/
set runtimepath^=~/.local/share/nvim/site/pack/packer/opt/plenary.nvim/

function! ActiveWinbar()
    return "hello"
endfunction

augroup Winbar
	autocmd!
	autocmd WinEnter,BufEnter * setlocal winbar=%!ActiveWinbar()
augroup END

set winbar=%!ActiveWinbar()
```

1. Open nvim with `nvim -u init.vim`.
1. Start telescope with `:Telescope`.

### Expected behavior

Only Telescope's window shown.

### Actual behavior

1. The winbar's value appears everywhere on telescope's window.
1. Cannot see what is typed in the input field of telescope.

![image](https://user-images.githubusercontent.com/16366403/170039158-653a2015-78bd-4883-b287-7d1c05fcfb62.png)

**Comment by ChrisAmelia** [comment]
@zeertzjq Duplicate of #18715 ?

**Comment by zeertzjq** [comment]
I don't think there is anything that can be changed in Nvim here, since you are explicitly setting winbar for every window. The only solution I can think of is to check if you actually want a winbar on the window before setting it. After #18773 global window will never apply to floating windows, so you can also use `setglobal winbar=%!ActiveWinbar()` instead.
